modules/index/src/main/java/org/apache/ignite/internal/index/ChangeIndexStatusTask.java

```diff
@@ -333,8 +335,26 @@ private <T> CompletableFuture<T> inBusyLocks(Supplier<CompletableFuture<T>> supp
 
         try {
             return supplier.get();
+        } catch (Throwable t) {
+            return failedFuture(t);
         } finally {
             leaveBusy();
         }
     }
+
+    private MetaIndexStatusChange statusChange() {
+        IndexMeta indexMeta = indexMetaStorage.indexMeta(indexDescriptor.id());
+
+        if (indexMeta == null) {
+            // Index was destroyed under a low watermark, well, we need to build it.
+            throw new IndexTaskStoppingException();
+        }
+
+        MetaIndexStatusChange statusChange = indexMeta.statusChangeNullable(MetaIndexStatus.convert(indexDescriptor.status()));
+
+        assert statusChange != null : IgniteStringFormatter.format("Missing index status change: [indexId={}, catalogStatus={}]",
```

Comment:
What if we run without assertions? We'll just return null and get an NPE further. How about using `Objects#requireNonNull()` instead of asserting? This doesn't seem to be a hot code.

Reply:
Yes, this is not a hot code and it shouldn’t be, so let’s leave it as an assertion for now.

Reply:
This is not a hot code, so we can check for null properly at each invocation and not use assertion for this.

It's much better to see a `null` right away than deal with an NPE later on.